Merged external modules break protected member access checking

Having some difficulty utilizing the external module augmentations feature merged in #6213. Here is a test case to demonstrate the problem:

```
// a.ts
export class A {
    protected protected: any;
    
    public setProtected(val: any) {
        this.protected = val;
    }
}
```

```
// b.ts
import {A} from './a'; // Only here to make b.ts an external module

declare module "./a" {
    interface A { }
}
```

This results in the following error:

> a.ts(5,9): error TS2446: Property 'protected' is protected and only accessible through an instance of class 'A'.

It looks like the problem is that [`checkClassPropertyAccess`](https://github.com/Microsoft/TypeScript/blob/1ae7d5ca092d883213f0a5f10c609fda7fa5cccc/src/compiler/checker.ts#L8777) in checker.ts ends up obtaining the merged interface as the enclosing class of the property access, rather than the actual enclosing class, then tries to ensure that this interface is derived from the class which declares the property, which isn't the case.

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -8776,7 +8776,7 @@ namespace ts {
          */
         function checkClassPropertyAccess(node: PropertyAccessExpression | QualifiedName, left: Expression | QualifiedName, type: Type, prop: Symbol): boolean {
             const flags = getDeclarationFlagsFromSymbol(prop);
-            const declaringClass = <InterfaceType>getDeclaredTypeOfSymbol(prop.parent);
+            const declaringClass = <InterfaceType>getDeclaredTypeOfSymbol(getMergedSymbol(prop.parent));
 
             if (left.kind === SyntaxKind.SuperKeyword) {
                 const errorNode = node.kind === SyntaxKind.PropertyAccessExpression ?

--- a/tests/baselines/reference/declarationMerging1.js
+++ b/tests/baselines/reference/declarationMerging1.js
@@ -1,0 +1,21 @@
+//// [tests/cases/compiler/declarationMerging1.ts] ////
+
+//// [file1.ts]
+class A {
+    protected _f: number;
+    getF() { return this._f; }
+}
+
+//// [file2.ts]
+interface A {
+    run();
+}
+
+//// [file1.js]
+var A = (function () {
+    function A() {
+    }
+    A.prototype.getF = function () { return this._f; };
+    return A;
+}());
+//// [file2.js]

--- a/tests/baselines/reference/declarationMerging1.symbols
+++ b/tests/baselines/reference/declarationMerging1.symbols
@@ -1,0 +1,21 @@
+=== tests/cases/compiler/file1.ts ===
+class A {
+>A : Symbol(A, Decl(file1.ts, 0, 0), Decl(file2.ts, 0, 0))
+
+    protected _f: number;
+>_f : Symbol(_f, Decl(file1.ts, 0, 9))
+
+    getF() { return this._f; }
+>getF : Symbol(getF, Decl(file1.ts, 1, 25))
+>this._f : Symbol(_f, Decl(file1.ts, 0, 9))
+>this : Symbol(A, Decl(file1.ts, 0, 0), Decl(file2.ts, 0, 0))
+>_f : Symbol(_f, Decl(file1.ts, 0, 9))
+}
+
+=== tests/cases/compiler/file2.ts ===
+interface A {
+>A : Symbol(A, Decl(file1.ts, 0, 0), Decl(file2.ts, 0, 0))
+
+    run();
+>run : Symbol(run, Decl(file2.ts, 0, 13))
+}

--- a/tests/baselines/reference/declarationMerging1.types
+++ b/tests/baselines/reference/declarationMerging1.types
@@ -1,0 +1,21 @@
+=== tests/cases/compiler/file1.ts ===
+class A {
+>A : A
+
+    protected _f: number;
+>_f : number
+
+    getF() { return this._f; }
+>getF : () => number
+>this._f : number
+>this : this
+>_f : number
+}
+
+=== tests/cases/compiler/file2.ts ===
+interface A {
+>A : A
+
+    run();
+>run : () => any
+}

--- a/tests/baselines/reference/declarationMerging2.js
+++ b/tests/baselines/reference/declarationMerging2.js
@@ -1,0 +1,32 @@
+//// [tests/cases/compiler/declarationMerging2.ts] ////
+
+//// [a.ts]
+
+export class A {
+    protected _f: number;
+    getF() { return this._f; }
+}
+
+//// [b.ts]
+export {}
+declare module "./a" {
+    interface A {
+        run();
+    }
+}
+
+//// [a.js]
+define(["require", "exports"], function (require, exports) {
+    "use strict";
+    var A = (function () {
+        function A() {
+        }
+        A.prototype.getF = function () { return this._f; };
+        return A;
+    }());
+    exports.A = A;
+});
+//// [b.js]
+define(["require", "exports"], function (require, exports) {
+    "use strict";
+});

--- a/tests/baselines/reference/declarationMerging2.symbols
+++ b/tests/baselines/reference/declarationMerging2.symbols
@@ -1,0 +1,25 @@
+=== tests/cases/compiler/a.ts ===
+
+export class A {
+>A : Symbol(A, Decl(a.ts, 0, 0), Decl(b.ts, 1, 22))
+
+    protected _f: number;
+>_f : Symbol(_f, Decl(a.ts, 1, 16))
+
+    getF() { return this._f; }
+>getF : Symbol(getF, Decl(a.ts, 2, 25))
+>this._f : Symbol(_f, Decl(a.ts, 1, 16))
+>this : Symbol(A, Decl(a.ts, 0, 0), Decl(b.ts, 1, 22))
+>_f : Symbol(_f, Decl(a.ts, 1, 16))
+}
+
+=== tests/cases/compiler/b.ts ===
+export {}
+declare module "./a" {
+    interface A {
+>A : Symbol(A, Decl(a.ts, 0, 0), Decl(b.ts, 1, 22))
+
+        run();
+>run : Symbol(run, Decl(b.ts, 2, 17))
+    }
+}

--- a/tests/baselines/reference/declarationMerging2.types
+++ b/tests/baselines/reference/declarationMerging2.types
@@ -1,0 +1,25 @@
+=== tests/cases/compiler/a.ts ===
+
+export class A {
+>A : A
+
+    protected _f: number;
+>_f : number
+
+    getF() { return this._f; }
+>getF : () => number
+>this._f : number
+>this : this
+>_f : number
+}
+
+=== tests/cases/compiler/b.ts ===
+export {}
+declare module "./a" {
+    interface A {
+>A : A
+
+        run();
+>run : () => any
+    }
+}

--- a/tests/cases/compiler/declarationMerging1.ts
+++ b/tests/cases/compiler/declarationMerging1.ts
@@ -1,0 +1,10 @@
+// @filename: file1.ts
+class A {
+    protected _f: number;
+    getF() { return this._f; }
+}
+
+// @filename: file2.ts
+interface A {
+    run();
+}

--- a/tests/cases/compiler/declarationMerging2.ts
+++ b/tests/cases/compiler/declarationMerging2.ts
@@ -1,0 +1,15 @@
+// @module: amd
+
+// @filename: a.ts
+export class A {
+    protected _f: number;
+    getF() { return this._f; }
+}
+
+// @filename: b.ts
+export {}
+declare module "./a" {
+    interface A {
+        run();
+    }
+}

